indico/modules/rb_new/client/js/common/timeline/SingleRoomTimelineModal.jsx

```diff
@@ -0,0 +1,137 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2019 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import React, {useEffect} from 'react';
+import moment from 'moment';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import PropTypes from 'prop-types';
+import {Modal, Icon, Dimmer, Loader, Popup} from 'semantic-ui-react';
+
+import {actions as bookRoomActions, selectors as bookRoomSelectors} from '../../modules/bookRoom';
+import TimelineLegend from './TimelineLegend';
+import DailyTimelineContent from './DailyTimelineContent';
+
+import './SingleRoomTimelineModal.module.scss';
+
+
+const _getRowSerializer = (day, room) => {
+    return ({bookings, preBookings, candidates, conflictingCandidates, nonbookablePeriods, unbookableHours,
+             blockings, conflicts, preConflicts}) => ({
+        availability: {
+            candidates: (candidates[day] || []).map((candidate) => (
+                {...candidate, bookable: false})
+            ) || [],
+            conflictingCandidates: (conflictingCandidates[day] || []).map((candidate) => (
+                {...candidate, bookable: false}
+            )) || [],
+            preBookings: preBookings[day] || [],
+            bookings: bookings[day] || [],
+            conflicts: conflicts[day] || [],
+            preConflicts: preConflicts[day] || [],
+            nonbookablePeriods: nonbookablePeriods[day] || [],
+            unbookableHours: unbookableHours || [],
+            blockings: blockings[day] || []
+        },
+        label: moment(day).format('L'),
+        key: day,
+        conflictIndicator: true,
+        room
+    });
+};
+
+const _SingleRoomTimelineContent = props => {
+    const {availability, availabilityLoading, room, filters, actions: {fetchAvailability}} = props;
+    useEffect(() => {
+        fetchAvailability(room, filters);
+    }, [fetchAvailability, filters, room]);
+
+    const isLoaded = !_.isEmpty(availability) && !availabilityLoading;
+    const dimmer = <Dimmer active page styleName="dimmer"><Loader /></Dimmer>;
```

Comment:
```jsx
if (!isLoaded) {
    return <Dimmer active page styleName="dimmer"><Loader /></Dimmer>;
}
```

Reply:
I don't understand this. You meant for the return line?